### DISCUSSION TITLE
Random available Rides like in RCT1

### DISCUSSION THREAD

**Discussion by TheLarry98** [OpenRCT2/OpenRCT2#15705] [Feature requests]
I don't know if you know it but this is something which is important for some scenarios from RCT1.
In RCT1 there were some scenarios where the game randomly decided which attraction you get and which you can research.
e.g. in the first Park: Forest Frontier from RCT1 the game decides if you are getting the Miniature Railway OR the Monorail (Small and Streamlined Monorail Trains) If you are getting the Miniature Railway than you have to research the monorail ((Small and Streamlined Monorail Trains)) and the other way around. This was also implemented in RCTC (or RTCC which some guys call it :P) so it should work in the rct2 engine.

**Comment by Gymnasiast** [comment]
This is a printout of Forest Frontiers' research:
```
IStarting with pre-invented
Theme: 0
Vehicle: 0 (15)
Vehicle: 2 (0)
Vehicle: 3 (3)
Vehicle: 4 (2)
Vehicle: 5 (4)
Vehicle: 6 (1)
Vehicle: 8 (7)
Vehicle: 10 (6)
Vehicle: 12 (8)
Vehicle: 13 (5)
Vehicle: 14 (9)
Vehicle: 18 (10)
Vehicle: 19 (11)
Vehicle: 20 (7)
Vehicle: 21 (9)
Vehicle: 23 (6)
Vehicle: 25 (12)
Vehicle: 26 (13)
Vehicle: 27 (16)
Vehicle: 28 (14)
Vehicle: 29 (17)
Vehicle: 30 (18)
Vehicle: 31 (19)
Vehicle: 32 (10)
Vehicle: 33 (11)
Vehicle: 34 (11)
Vehicle: 35 (22)
Vehicle: 36 (24)
Vehicle: 37 (23)
Vehicle: 38 (25)
Vehicle: 39 (26)
Vehicle: 40 (27)
Vehicle: 41 (33)
Vehicle: 42 (37)
Vehicle: 43 (38)
Vehicle: 44 (39)
Vehicle: 45 (40)
Vehicle: 46 (41)
Vehicle: 47 (42)
Vehicle: 48 (44)
Vehicle: 50 (46)
Vehicle: 51 (47)
Ride: 47
Ride: 21
Ride: 33
Ride: 11
Ride: 8
Ride: 5
Ride: 0
Ride: 4
Ride: 46
Ride: 26
Ride: 36
Ride: 30
Ride: 32
Ride: 28
End of pre-invented, starting to-be-researched
Ride: 37
Ride: 23
Ride: 35
Ride: 15
Ride: 14
Ride: 20
Ride: 2
Vehicle: 7 (4)
Ride: 10
Ride: 29
Vehicle: 52 (4)
Ride: 9
Ride: 25
Ride: 34
Ride: 16
Ride: 18
Ride: 31
Ride: 6
Ride: 1
Ride: 48
Ride: 45
Ride: 41
End of to-be-researched, starting random
Ride: 5
Ride: 6
Ride: 9
Ride: 2
Ride: 28
Ride: 31
Ride: 48
Ride: 31
Ride: 45
Ride: 29
Vehicle: 7 (4)
Vehicle: 52 (4)
Ride: 25
Ride: 20
Ride: 23
Ride: 15
```
Ride 5 = miniature railway
Ride 6 = monorail

Basically, it seems that, despite the items already existing earlier in the research list, the random list overrides that. That explains why the Miniature Railway is always available if you open the scenario in OpenRCT2.
